src/min-heap-api.c

```diff
@@ -0,0 +1,202 @@
+    if (heap == NULL || heap->compare == NULL)
+        return MIN_HEAP_NULL_POINTER;
+    if (heap->size == 0)
+        return MIN_HEAP_EMPTY;
+    if (index >= heap->size)
+        return MIN_HEAP_OUT_OF_BOUNDS;
+
+    // Swap the error with the last one in the heap (if not the same)
+    uint8_t *base = (uint8_t *)heap->data;
+    const size_t data_size = heap->data_size;
+    if (heap->size > 1)
+        _min_heap_swap(heap, base + index * data_size, base + (heap->size - 1) * data_size);
+
+    // Remove last element
+    --heap->size;
+
+    // Copy element
+    if (out != NULL)
+        memcpy(out, base + heap->size * data_size, data_size);
+
+    if (index == heap->size)
+        return MIN_HEAP_OK;
+
+    // Restore heap properties
+    int8_t cmp = heap->compare(base + index * data_size, base + heap->size * data_size);
+    // Up-heapify
+    if (cmp < 0) {
+        size_t parent = MIN_HEAP_PARENT(index);
+        while (index != 0 && heap->compare(base + index * data_size, base + parent * data_size) < 0) {
+            // Swap items
+            _min_heap_swap(heap, base + index * data_size, base + parent * data_size);
+
+            // Update indices
+            index = parent;
+            parent = MIN_HEAP_PARENT(index);
+        }
+    }
+    // Down-heapify
+    else if (cmp > 0) {
+        // Restore heap properties
+        size_t l = MIN_HEAP_CHILD_L(index);
+        size_t r = MIN_HEAP_CHILD_R(index);
+        size_t child = (r >= heap->size)
+                           ? l
+                           : (heap->compare(base + l * data_size, base + r * data_size) < 0
+                                  ? l
+                                  : r);
+
+        // Until a leaf is reached (or the parent has only the left child)
+        while (r < heap->size && heap->compare(base + child * data_size, base + index * data_size) < 0) {
+            _min_heap_swap(heap, base + index * data_size, base + child * data_size);
+
+            // Update indices
+            index = child;
+            l = MIN_HEAP_CHILD_L(index);
+            r = MIN_HEAP_CHILD_R(index);
+            child = (r >= heap->size)
+                        ? l
+                        : (heap->compare(base + l * data_size, base + r * data_size) < 0
+                               ? l
+                               : r);
+        }
+
+        // Check left child
+        if (l < heap->size && heap->compare(base + l * data_size, base + index * data_size) < 0)
+            _min_heap_swap(heap, base + index * data_size, base + child * data_size);
+    }
+    return MIN_HEAP_OK;
+}
+
+signed_size_t _min_heap_find(MinHeapInterface *heap, void *item) {
+    if (heap == NULL || item == NULL || heap->compare == NULL || heap->size == 0 || heap->data == NULL)
+        return -1;
+
+    for (size_t i = 0; i < heap->size; ++i) {
+        if (heap->compare(item, heap->data + heap->data_size * i) == 0)
```

Comment:
`warning: pointer of type 'void *' used in arithmetic [-Wpointer-arith]`
```suggestion
        if (heap->compare(item, (uint8_t *)heap->data + heap->data_size * i) == 0)
```